sql/core/src/main/scala/org/apache/spark/sql/streaming/DataStreamWriter.scala

```diff
@@ -235,6 +237,21 @@ final class DataStreamWriter[T] private[sql](ds: Dataset[T]) {
         "write files of Hive data source directly.")
     }
 
+    val hadoopConf = df.sparkSession.sessionState.newHadoopConf()
+    val defaultFS = FileSystem.getDefaultUri(hadoopConf).getScheme
+    val tmpFS = new URI(System.getProperty("java.io.tmpdir")).getScheme
+
+    val isTempCheckpointLocationAvailable = tmpFS match {
+      case null | "file" =>
+        if (defaultFS == null || defaultFS.equals("file")) {
+          true
+        } else {
+          false
+        }
+      case defaultFS => true
```

Comment:
@mgaido91 , if defaultFS is HDFS, in which scenario `tmpFS` ("java.io.tmpdir") be the same as `defaultFS`? From my understanding, unless we change "java.io.tmpdir" property, then `tmpFs` will always be local FS.